frontend/packages/console-app/src/components/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -0,0 +1,279 @@
+        setInProgress(false);
+      });
+  };
+
+  const title = 'Create VolumeSnapshot';
+  return (
+    <div className="co-m-pane__body co-volume-snapshot__body">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <form className="co-m-pane__body-group co-m-pane__form" onSubmit={create}>
+        <h1 className="co-m-pane__heading">{title}</h1>
+        {kindObj.kind === VolumeSnapshotModel.kind && (
+          <PVCDropdown
+            namespace={namespace}
+            onChange={setPVCName}
+            id="claimName"
+            selectedKey={pvcName}
+          />
+        )}
+        {kindObj.kind === PersistentVolumeClaimModel.kind && (
+          <p>
+            Creating snapshot for claim <strong>{resourceName}</strong>
+          </p>
+        )}
+        {pvcStatus && pvcStatus !== 'Bound' && (
+          <Alert
+            className="co-alert"
+            variant="warning"
+            title="Snapshot creation for unbound claim is not recommended."
+            isInline
+          />
+        )}
+        <div className="form-group co-volume-snapshot__form">
+          <label className="control-label co-required" htmlFor="snapshot-name">
+            Name
+          </label>
+          <input
+            className="pf-c-form-control"
+            type="text"
+            onChange={handleSnapshotName}
+            name="snapshotName"
+            id="snapshot-name"
+            value={snapshotName}
+            required
+          />
+        </div>
+        <div className="form-group co-volume-snapshot__form">
+          <label className="control-label co-required" htmlFor="snapshot-class">
+            Snapshot Class
+          </label>
+          <SnapshotClassDropdown
+            onChange={setSnapshotClassName}
+            id="claimName"
+            selectedKey={snapshotClassName}
+            pvcProvisioner={pvcProvisioner}
+          />
+        </div>
+        <ButtonBar errorMessage={error} inProgress={inProgress}>
+          <ActionGroup className="pf-c-form">
+            <Button
+              type="submit"
+              variant="primary"
+              id="save-changes"
+              isDisabled={!snapshotClassName || !snapshotName || !pvcName || pvcStatus !== 'Bound'}
+            >
+              Create
+            </Button>
+            <Button type="button" variant="secondary" onClick={history.goBack}>
+              Cancel
+            </Button>
+          </ActionGroup>
+        </ButtonBar>
+      </form>
+      <div className="co-volume-snapshot__info">
+        <Grid hasGutter>
+          <GridItem span={2} />
+          <GridItem span={8}>
+            <PVCSummary
+              namespace={namespace}
+              status={pvcStatus}
+              pvc={pvcName}
+              persistentVolumeClaim={pvcObj}
+            />
+          </GridItem>
+          <GridItem span={2} />
+        </Grid>
+      </div>
+    </div>
+  );
+};
+
+// eslint-disable-next-line no-underscore-dangle
```

Comment:
you should not disable the rule
the VolumeSnapshot_ can be written without dongle (and the current one is also missing types)

```suggestion
export const VolumeSnapshot = connectToPlural<VolumeSnapshotProps>(
  ({ kindObj, kindsInFlight, match: { params } }) => {
    if (!kindObj && kindsInFlight) {
      return <LoadingBox />;
    }
    return (
      <CreateSnapshotForm namespace={params.ns} resourceName={params.name} kindObj={kindObj} />
    );
  },
);

type VolumeSnapshotProps = WithPluralProps & {
  match: match<{ name: string; ns: string }>;
};
```